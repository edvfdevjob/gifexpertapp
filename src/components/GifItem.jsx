
import PropTypes from 'prop-types';

export const GifItem = ({ title, url, id }) => {

  return (
    <div className="card">

        <img src={ url } alt={ title } />
        <p>{ title }</p>

    </div>
  )
}

/* Agregar PropTypes
  a. title obligatorio ??? yarn add
  b.url obligatorio

2. Evaluar el snapshot


*/

GifItem.propTypes = {
  title: PropTypes.string.isRequired, 
  url: PropTypes.string.isRequired
}