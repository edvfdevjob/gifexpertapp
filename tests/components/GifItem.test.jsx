import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem', () => { 

    const title = 'Saitam';
    const url = 'https://facebook.com/';


    test('Evaluar el snapshot de GifItem', () => { 

        
        const { container } = render( <GifItem title={ title } url={ url } /> );

        expect( container ).toMatchSnapshot(); 

     });

     test('debe de mostrar la imagen con el url y alt indicado', () => { 

        render( <GifItem title={ title } url={ url } /> );
        // screen.debug(); 
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img'); 
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

      });

      test('debe de mostrar el titulo en el componente', () => { 

        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();

       })

 })