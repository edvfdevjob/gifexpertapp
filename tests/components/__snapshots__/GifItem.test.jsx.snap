// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Evaluar el snapshot de GifItem 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitam"
      src="https://facebook.com/"
    />
    <p>
      Saitam
    </p>
  </div>
</div>
`;
